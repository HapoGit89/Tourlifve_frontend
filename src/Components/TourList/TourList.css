.TourList {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    margin-left: 5%;
    margin-right:5%;
    flex-wrap:wrap;
}

.TourCard {
 margin: 2%;

}


.AddTourButton{
    margin-top: 5%;
 
}


.EmptyTourList {
    color:whitesmoke;
    font-size: x-large;
    margin: 5%;
}