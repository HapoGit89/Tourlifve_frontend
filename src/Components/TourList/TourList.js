import {React, useContext} from "react"
import "./TourList.css"
import userContext from "../../userContext"
import TourCard from "../TourCard/TourCard"
import { Button } from "reactstrap"
import { useNavigate } from "react-router-dom"


function TourList(){
  const user = useContext(userContext)
  const navigate = useNavigate()


    if (user.token){  //Route protection
    return(
      <div className="TourListPage">
            
      <Button className="AddTourButton"
        color="secondary"
        onClick={()=>navigate("./new")}
      >
        Create NEW ...
      </Button>

        <div className="TourList">
          
          {user.tours.map((el,id)=>(<div className="TourCard" key={id}><TourCard tour={el}/></div>))}
        </div>
        </div>
    )}

    else{
      return (
        <div>
          <h1>Please log in to see this page</h1>
        </div>
      )
    }
}

export default TourList