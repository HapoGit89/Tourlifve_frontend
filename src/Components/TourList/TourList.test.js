import { render,screen } from '@testing-library/react';
import TourList from './TourList';
import userContext from '../../userContext';
import { BrowserRouter } from 'react-router-dom';



// Smoke Test

it('renders without crashing', () => {
    const user = {username: "hannes", token: "ajshajhjshajhjhs", tours: []}
    render(<BrowserRouter>
            <userContext.Provider value={user}>
            <TourList/>
            </userContext.Provider>
           </BrowserRouter>
      )
});


// Snapshot Test

it('matches Snapshot', () => {
    const user = {username: "hannes", token: "ajshajhjshajhjhs", tours: []}
    const {asFragement} = render(<BrowserRouter>
            <userContext.Provider value={user}>
            <TourList/>
            </userContext.Provider>
           </BrowserRouter>
      )
      expect(asFragement).toMatchSnapshot()
});

//

it('shows add button', () => {
    const user = {username: "hannes", token: "ajshajhjshajhjhs", tours: []}
    const {} = render(<BrowserRouter>
            <userContext.Provider value={user}>
            <TourList/>
            </userContext.Provider>
           </BrowserRouter>
      )

      expect(screen.getByText("Create New Tour ...")).toBeInTheDocument()
     
});