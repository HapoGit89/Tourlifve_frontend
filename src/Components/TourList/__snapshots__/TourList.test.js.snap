// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `undefined`;

exports[`show add button 1`] = `undefined`;
